File: src/network/packet/impl/response/PlayerWhitelistCheckResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response;

use pocketcloud\cloud\network\packet\ResponsePacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class PlayerWhitelistCheckResponsePacket extends ResponsePacket {

    public function __construct(private readonly bool $whitelisted = false) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->whitelisted);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function isWhitelisted(): bool {
        return $this->whitelisted;
    }

    public static function create(bool $whitelisted): self {
        return new self($whitelisted);
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/response/PlayerWhitelistCheckResponsePacket.java
package de.pocketcloud.cloud.network.packet.impl.response;

import de.pocketcloud.cloud.network.packet.ResponsePacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerWhitelistCheckResponsePacket extends ResponsePacket {

    private boolean whitelisted;

    public PlayerWhitelistCheckResponsePacket(boolean whitelisted) {
        this.whitelisted = whitelisted;
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(whitelisted);
    }

    public static PlayerWhitelistCheckResponsePacket create(boolean whitelisted) {
        return new PlayerWhitelistCheckResponsePacket(whitelisted);
    }
}
